src/SFA.DAS.EAS.Employer_Financial.Database/Scripts/Manual/GenerateTestData/AdHoc/CreateLevies.sql

```diff
@@ -0,0 +1,146 @@
+			ELSE
+				@monthlyLevy*dbo.PayrollMonth(DATEADD(month,/*monthBeforeToDate*/ -1-@numberOfMonthsToCreate+ROW_NUMBER() OVER (ORDER BY [object_id]),@toDate))
+			END),
+			DATEADD(month,/*monthBeforeToDate*/ -@numberOfMonthsToCreate+ROW_NUMBER() OVER (ORDER BY [object_id]),@toDate),
+			dbo.PayrollYear(DATEADD(month,/*monthBeforeToDate*/ -1-@numberOfMonthsToCreate+ROW_NUMBER() OVER (ORDER BY [object_id]),@toDate)),
+			dbo.PayrollMonth(DATEADD(month,/*monthBeforeToDate*/ -1-@numberOfMonthsToCreate+ROW_NUMBER() OVER (ORDER BY [object_id]),@toDate))
+FROM sys.all_objects
+ORDER BY monthBeforeToDate;
+
+--*** Levy Adjustments ***
+--declare @adjustmentAmount DECIMAL(18, 4) = 1000
+--update @levyDecByMonth set amount = amount-@monthlyLevy-@adjustmentAmount where payrollYear = '18-19' and payrollMonth >= 6
+
+select * from @levyDecByMonth
+
+---
+--- Generate english fraction rows to cover the levy decs we're about to generate
+---
+
+-- engligh fractions usually generated on 1, 4, 7, 10 (not 3, 6, 9, 12), but can be generated anytime
+declare @englishFractionMonths table (dateCalculated DATETIME)
+
+-- first quarter before (or on) first month
+insert @englishFractionMonths
+select top 1 dateadd(month,-datepart(month,createMonth)%3, createMonth) from @levyDecByMonth order by createMonth
+
+-- rest of the quarters
+insert @englishFractionMonths
+select createMonth from (select createMonth from @levyDecByMonth except select top 1 createMonth from @levyDecByMonth order by createMonth) x where datepart(month,createMonth)%3 = 0
+
+-- only insert english fraction rows that don't already exist (and add english fraction calcs on consistent day of the month)
+declare @newEnglishFractionMonths table (dateCalculated DATETIME)
+
+insert @newEnglishFractionMonths
+select datefromparts(datepart(year,dateCalculated), datepart(month,dateCalculated), 7) from @englishFractionMonths
+except select dateCalculated from employer_financial.EnglishFraction where EmpRef = @payeScheme
+
+INSERT employer_financial.EnglishFraction (DateCalculated, Amount, EmpRef, DateCreated)
+select dateCalculated, 1.0, @payeScheme, dateCalculated from @newEnglishFractionMonths
+
+---
+--- Generate levy decs
+---
+
+DECLARE @maxSubmissionId BIGINT = ISNULL((SELECT MAX(SubmissionId) FROM employer_financial.levydeclaration),0)
+
+declare @baselineSubmissionDate datetime = datefromparts(year(@toDate), month(@toDate), 18)
+declare @baselineCreatedDate datetime = datefromparts(year(@toDate), month(@toDate), 20)
+declare @baselinePayrollDate datetime = DATEADD(month, -1, @toDate)
+
+--todo use monthBeforeToDate, rather than row_number?
+INSERT INTO employer_financial.levydeclaration (AccountId,empref,levydueytd,levyallowanceforyear,submissiondate,submissionid,payrollyear,payrollmonth,createddate,hmrcsubmissionid)
+select @accountId, @payeScheme, 
+	amount,
+	1500.0000, 
+	DATEADD(month, monthBeforeToDate, @baselineSubmissionDate), 
+	@maxSubmissionId + row_number() over (order by (select NULL)), 
```

Comment:
I've changed it to use the existing `monthBeforeToDate` column instead of the `ROW_NUMBER OVER`s. 

~~Yes it could (I tried it and it worked), however, as it only allows us to tidy up the two 'row_number over's in this statement, against adding a column to the source table, I think it better to have slightly more complicated implementation vs keeping the user interface for the testers as simple as possible.~~

~~If we could somehow leverage an identity column within `GenerateLevySourceTable` to remove its 'row_number over`'s, I think that would tilt the balance in favour of using an identity column.~~